javascript-client/src/client.ts

```diff
@@ -47,12 +47,18 @@ export interface ProgressEvent {
 type ProgressCallback = (progress: ProgressEvent) => void;
 
 export default class S3FFClient {
+  protected readonly api: AxiosInstance;
+
   constructor(
-    protected readonly baseUrl: string,
-    private readonly onProgress: ProgressCallback = () => { /* no-op */ },
+    baseUrl: string,
+    protected readonly onProgress: ProgressCallback = () => { /* no-op */ },
+    apiConfig: AxiosRequestConfig = {},
```

Comment:
Another question is whether this should take an options object, so `apiConfig` could be passed without needing to specify `onProgress`.

In the event that downstreams want to pass both `apiConfig` and `onProgress`, it's slightly more verbose to have to pass the labeled options object, rather than just positional arguments.

I think most downstreams who want to pass an `apiConfig` will at least care about progress too, but that's just speculation.

Finally, it means we can't use the handy concise [parameter properties](https://www.typescriptlang.org/docs/handbook/classes.html#parameter-properties) to also declare the member in the same place.

Reply:
I prefer an opts Object rather than positional args for anything that isn't strictly required to be passed. I dislike that parameters properties syntax anyway, so I'm fine to lose it.